charts/kyverno/values.yaml

```diff
@@ -216,11 +216,28 @@ config:
   - '[TokenReview,*,*]'
   - '[SubjectAccessReview,*,*]'
   - '[SelfSubjectAccessReview,*,*]'
-  - '[*,{{ include "kyverno.namespace" . }},kyverno*]'
   - '[Binding,*,*]'
   - '[ReplicaSet,*,*]'
   - '[ReportChangeRequest,*,*]'
   - '[ClusterReportChangeRequest,*,*]'
+  # exclude resources from the chart
+  - '[ClusterRole,*,{{ template "kyverno.fullname" . }}:*]'
+  - '[ClusterRoleBinding,*,{{ template "kyverno.fullname" . }}:*]'
+  - '[ServiceAccount,{{ include "kyverno.namespace" . }},{{ template "kyverno.serviceAccountName" . }}]'
+  - '[ConfigMap,{{ include "kyverno.namespace" . }},{{ template "kyverno.configMapName" . }}]'
+  - '[ConfigMap,{{ include "kyverno.namespace" . }},{{ template "kyverno.metricsConfigMapName" . }}]'
+  - '[Deployment,{{ include "kyverno.namespace" . }},{{ template "kyverno.fullname" . }}]'
```

Comment:
IMHO, we should exclude Kyverno deployment.

Will there be any issue if we don't exclude other Kyverno resources?

Reply:
As @treydock says, I think it depends of the policies installed in the cluster.
By default excluding the deployment should be enough but depending on the policies installed it could be not enough.

Not sure what we want, harder maintenance but better security or easier maintenance but relaxed security.
In any case I think we should at least add some notes in the docs.

Reply:
Any comment @JimBugwadia @chipzoller ?

Reply:
I was wondering if during the initial helm installation there's a way to concatenate all the Kyverno-specific `resourceFilters` defaults with the user-defined value in the values file. Please correct me if I'm wrong, but looking at the way it is implemented right now I believe the user-provided `resourceFilter` value will override the defaults defined here.

_(Please ignore if it's unnecessary and complicates things further)_

Reply:
No, it would require another value like `additionalResourceFilters`.

Reply:
In that case I think we should add a note for the user in the installation docs to make sure they're aware of this.
Another way to handle this by default could be that we store these defaults in a separate variable, and then concatenate them with the `resourceFilters`.

Reply:
> Another way to handle this by default could be that we store these defaults in a separate variable, and then concatenate them with the resourceFilters.

And it would be appropriate to perform this concatenation within the resource manifests, and not on the variables themselves.

Reply:
The chart docs definitely need to be improved.

Reply:
@JimBugwadia @chipzoller ?

Reply:
@eddycharly - apologies, but I missed this earlier. The Kyverno install should be secure by default, and seems like allowing namespace level exclusions would not be sufficient from a security perspective. I am not sure what the solution is - we probably need a more detailed discussion on this.

Reply:
@JimBugwadia I played a bit with object selector (https://github.com/kyverno/kyverno/pull/3413). Used carefully it could help IMHO.
In any case, I feel like kyveno alone will never bring a completely secure setup, we probably need to clarify what we mean by secure (secure in terms of malicious activity ? in terms of operations ? etc...).